File: src/main/java/com/uiptv/model/Configuration.java
package com.uiptv.model;

import com.uiptv.shared.BaseJson;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class Configuration extends BaseJson {
    private String dbId;
    private String playerPath1;
    private String playerPath2;
    private String playerPath3;
    private String defaultPlayerPath;
    private String filterCategoriesList;
    private String filterChannelsList;
    private String serverPort;
    private String cacheExpiryDays;
    private String languageLocale;
    private String tmdbReadAccessToken;
    private String uiZoomPercent;
    private String vlcNetworkCachingMs;
    private String vlcLiveCachingMs;
    private String publishedM3uCategoryMode;
    private boolean darkTheme;
    private boolean pauseFiltering;
    private boolean pauseCaching;
    private boolean embeddedPlayer;
    private boolean wideView;
    private boolean enableFfmpegTranscoding;
    private boolean enableLitePlayerFfmpeg;
    private boolean autoRunServerOnStartup;
    private boolean enableThumbnails = true;
    private boolean enableVlcHttpUserAgent = true;
    private boolean enableVlcHttpForwardCookies = true;
    private boolean resolveChainAndDeepRedirects = false;


    @SuppressWarnings("java:S107")
    public Configuration(String playerPath1, String playerPath2, String playerPath3, String defaultPlayerPath, String filterCategoriesList, String filterChannelsList, boolean pauseFiltering, boolean darkTheme, String serverPort, boolean embeddedPlayer, boolean enableFfmpegTranscoding, boolean enableThumbnails) {
        this.playerPath1 = playerPath1;
        this.playerPath2 = playerPath2;
        this.playerPath3 = playerPath3;
        this.defaultPlayerPath = defaultPlayerPath;
        this.filterCategoriesList = filterCategoriesList;
        this.filterChannelsList = filterChannelsList;
        this.pauseFiltering = pauseFiltering;
        this.darkTheme = darkTheme;
        this.serverPort = serverPort;
        this.embeddedPlayer = embeddedPlayer;
        this.enableFfmpegTranscoding = enableFfmpegTranscoding;
        this.enableThumbnails = enableThumbnails;
    }

    @SuppressWarnings("java:S107")
    public Configuration(String playerPath1, String playerPath2, String playerPath3, String defaultPlayerPath, String filterCategoriesList, String filterChannelsList, boolean pauseFiltering, boolean darkTheme, String serverPort, boolean embeddedPlayer, boolean enableFfmpegTranscoding) {
        this(playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, pauseFiltering, darkTheme, serverPort, embeddedPlayer, enableFfmpegTranscoding, true);
    }

    @SuppressWarnings("java:S107")
    public Configuration(String playerPath1, String playerPath2, String playerPath3, String defaultPlayerPath, String filterCategoriesList, String filterChannelsList, boolean pauseFiltering, boolean darkTheme, String serverPort, boolean embeddedPlayer, boolean enableFfmpegTranscoding, String cacheExpiryDays, boolean enableThumbnails) {
        this(playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, pauseFiltering, darkTheme, serverPort, embeddedPlayer, enableFfmpegTranscoding, enableThumbnails);
        this.cacheExpiryDays = cacheExpiryDays;
    }

    @SuppressWarnings("java:S107")
    public Configuration(String playerPath1, String playerPath2, String playerPath3, String defaultPlayerPath, String filterCategoriesList, String filterChannelsList, boolean pauseFiltering, boolean darkTheme, String serverPort, boolean embeddedPlayer, boolean enableFfmpegTranscoding, String cacheExpiryDays) {
        this(playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, pauseFiltering, darkTheme, serverPort, embeddedPlayer, enableFfmpegTranscoding, cacheExpiryDays, true);
    }

}

File: server/src/main/kotlin/com/uiptv/model/Configuration.kt
package com.uiptv.model

import com.uiptv.shared.BaseJson

data class Configuration @JvmOverloads constructor(
    var dbId: String? = null,
    var playerPath1: String? = null,
    var playerPath2: String? = null,
    var playerPath3: String? = null,
    var defaultPlayerPath: String? = null,
    var filterCategoriesList: String? = null,
    var filterChannelsList: String? = null,
    var serverPort: String? = null,
    var cacheExpiryDays: String? = null,
    var languageLocale: String? = null,
    var tmdbReadAccessToken: String? = null,
    var filterLockHash: String? = null,
    var filterLockUnlockDurationMinutes: String? = "15",
    var uiZoomPercent: String? = null,
    var vlcNetworkCachingMs: String? = null,
    var vlcLiveCachingMs: String? = null,
    var publishedM3uCategoryMode: String? = null,
    @get:JvmName("isDarkTheme")
    var darkTheme: Boolean = false,
    @get:JvmName("isPauseFiltering")
    var pauseFiltering: Boolean = false,
    @get:JvmName("isPauseCaching")
    var pauseCaching: Boolean = false,
    @get:JvmName("isEmbeddedPlayer")
    var embeddedPlayer: Boolean = false,
    @get:JvmName("isWideView")
    var wideView: Boolean = false,
    @get:JvmName("isEnableFfmpegTranscoding")
    var enableFfmpegTranscoding: Boolean = false,
    @get:JvmName("isEnableLitePlayerFfmpeg")
    var enableLitePlayerFfmpeg: Boolean = false,
    @get:JvmName("isAutoRunServerOnStartup")
    var autoRunServerOnStartup: Boolean = false,
    @get:JvmName("isEnableThumbnails")
    var enableThumbnails: Boolean = true,
    @get:JvmName("isEnableVlcHttpUserAgent")
    var enableVlcHttpUserAgent: Boolean = true,
    @get:JvmName("isEnableVlcHttpForwardCookies")
    var enableVlcHttpForwardCookies: Boolean = true,
    @get:JvmName("isResolveChainAndDeepRedirects")
    var resolveChainAndDeepRedirects: Boolean = false
) : BaseJson() {
    @Suppress("java:S107")
    constructor(
        playerPath1: String?,
        playerPath2: String?,
        playerPath3: String?,
        defaultPlayerPath: String?,
        filterCategoriesList: String?,
        filterChannelsList: String?,
        pauseFiltering: Boolean,
        darkTheme: Boolean,
        serverPort: String?,
        embeddedPlayer: Boolean,
        enableFfmpegTranscoding: Boolean,
        enableThumbnails: Boolean
    ) : this(null, playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, serverPort, null, null, null, null, "15", null, null, null, null, darkTheme, pauseFiltering, false, embeddedPlayer, false, enableFfmpegTranscoding, false, false, enableThumbnails, true, true, false)

    @Suppress("java:S107")
    constructor(
        playerPath1: String?,
        playerPath2: String?,
        playerPath3: String?,
        defaultPlayerPath: String?,
        filterCategoriesList: String?,
        filterChannelsList: String?,
        pauseFiltering: Boolean,
        darkTheme: Boolean,
        serverPort: String?,
        embeddedPlayer: Boolean,
        enableFfmpegTranscoding: Boolean
    ) : this(playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, pauseFiltering, darkTheme, serverPort, embeddedPlayer, enableFfmpegTranscoding, true)

    @Suppress("java:S107")
    constructor(
        playerPath1: String?,
        playerPath2: String?,
        playerPath3: String?,
        defaultPlayerPath: String?,
        filterCategoriesList: String?,
        filterChannelsList: String?,
        pauseFiltering: Boolean,
        darkTheme: Boolean,
        serverPort: String?,
        embeddedPlayer: Boolean,
        enableFfmpegTranscoding: Boolean,
        cacheExpiryDays: String?,
        enableThumbnails: Boolean
    ) : this(null, playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, serverPort, cacheExpiryDays, null, null, null, "15", null, null, null, null, darkTheme, pauseFiltering, false, embeddedPlayer, false, enableFfmpegTranscoding, false, false, enableThumbnails, true, true, false)

    @Suppress("java:S107")
    constructor(
        playerPath1: String?,
        playerPath2: String?,
        playerPath3: String?,
        defaultPlayerPath: String?,
        filterCategoriesList: String?,
        filterChannelsList: String?,
        pauseFiltering: Boolean,
        darkTheme: Boolean,
        serverPort: String?,
        embeddedPlayer: Boolean,
        enableFfmpegTranscoding: Boolean,
        cacheExpiryDays: String?
    ) : this(playerPath1, playerPath2, playerPath3, defaultPlayerPath, filterCategoriesList, filterChannelsList, pauseFiltering, darkTheme, serverPort, embeddedPlayer, enableFfmpegTranscoding, cacheExpiryDays, true)
}
